genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -479,6 +483,49 @@ func (g *GenesysCloudResourceExporter) generateOutputFiles() diag.Diagnostics {
 		}
 	}
 
+	if _, ok := g.d.GetOk("compress"); ok { //if true, compress directory name of where the export is going to occur
```

Comment:
Here is the description in the [docs for GetOk](https://pkg.go.dev/github.com/hashicorp/terraform/helper/schema#ResourceData.GetOk):

> GetOk returns the data for the given key and whether or not the key has been set to a non-zero value at some point.
>The first result will not necessarilly be nil if the value doesn't exist. The second result should be checked to determine this information.

I think the way you have it might actually work, since the second value indicates if `compress` has been set to a non zero value (which is true in the case of bools). Still the "at some point" confuses things a bit, and since the compress field has a `Default` value, there is no need to check if compress has been set to a value by the user.

I would use:

```golang
if compress := d.Get("compress").(bool); compress {
    // ...
}
```